qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,619 @@
+
+    The :class:`~.BackendEstimator` class is a generic implementation of the
+    :class:`~.BaseEstimator` interface that is used to wrap a :class:`~.Backend`
+    object in the :class:`~.BaseEstimator` API. It facilitates using backends
+    that do not provide a native :class:`~.BaseEstimator` implementation in
+    places that work with :class:`~.BaseEstimator`, such as algorithms in
+    :mod:`qiskit.algorithms` including :class:`~.qiskit.algorithms.minimum_eigensolvers.VQE`.
+    However, if you're using a provider that has a native implementation of
+    :class:`~.BaseEstimator`, it is a better choice to leverage that native
+    implementation as it will likely include additional optimizations and be a
+    more efficient implementation. The generic nature of this class precludes
+    doing any provider- or backend-specific optimizations.
+    """
+
+    # pylint: disable=missing-raises-doc
+    def __init__(
+        self,
+        backend: Backend,
+        *,  # TODO: allow backend as positional after removing deprecations
+        abelian_grouping: bool = True,
+        skip_transpilation: bool = False,
+        bound_pass_manager: PassManager | None = None,
+        options: dict | None = None,
+    ):
+        """Initalize a new BackendEstimator instance.
+
+        Args:
+            backend: Required: The backend to run the primitive on.
+            abelian_grouping: Whether commuting observable components should be grouped.
+            skip_transpilation: If `True`, transpilation of the input circuits is skipped.
+            bound_pass_manager: An optional pass manager to run after parameter binding.
+            options: Default options.
+        """
+        # TODO: validation
+        self.backend = backend
+        self.abelian_grouping = abelian_grouping
+        self.skip_transpilation = skip_transpilation  # TODO: tranpilation level
+        self._bound_pass_manager = bound_pass_manager  # TODO: standardize
+        super().__init__(
+            circuits=None,
+            observables=None,
+            parameters=None,
+            options=options,
+        )
+        self._transpile_options = Options()
+
+    ################################################################################
+    ## PROPERTIES
+    ################################################################################
+    @property
+    def backend(self) -> Backend:  # TODO: normalize to one type of Backend
+        """Backend to use for circuit measurements."""
+        return self._backend
+
+    @backend.setter
+    def backend(self, backend: Backend) -> None:
+        if not isinstance(backend, Backend):
+            raise TypeError(
+                "Expected `Backend` type for `backend`, " f"got `{type(backend)}` instead."
+            )
+        # TODO: clear all transpilation caching
+        if isinstance(backend, BackendV1):
+            backend.max_circuits = getattr(backend.configuration(), "max_experiments", None)
```

Comment:
This is a way of pseudo-normalizing the type of `backend` so that it looks like we are always using `BackendV2` internally and we avoid bloating other functions with checks based on the actual type.

Ideally we would want to do something like 
```python
backend = backendV1_to_V2(backend)
```

But I don't think this is possible, right?